test/Microsoft.AspNetCore.SignalR.Client.Tests/ConnectionTests.cs

```diff
@@ -406,9 +410,9 @@ public void CannotStartConnectionWithInvalidTransportType(TransportType requeste
             await connection.StartAsync();
             channel.Output.TryWrite(Array.Empty<byte>());
             channel.Output.TryWrite(Array.Empty<byte>());
```

Comment:
I think the second write needs to happen after the user's callback has been invoked.